src/test/scala/com/microsoft/hyperspace/index/IndexManagerTests.scala

```diff
@@ -294,6 +295,87 @@ class IndexManagerTests extends HyperspaceSuite with SQLHelper {
     }
   }
 
+  test("Verify quick optimize rebuild of index after index incremental refresh.") {
+    withTempPathAsString { testPath =>
+      val indexConfig = IndexConfig(s"index", Seq("RGUID"), Seq("imprs"))
+      import spark.implicits._
+      SampleData.testData
+        .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+        .limit(10)
+        .write
+        .parquet(testPath)
+      val df = spark.read.parquet(testPath)
+      hyperspace.createIndex(df, indexConfig)
+      val indexCount = spark.read
+        .parquet(s"$systemPath/index/${IndexConstants.INDEX_VERSION_DIRECTORY_PREFIX}=0")
+        .count()
+      assert(indexCount == 10)
+
+      // Change original data.
+      SampleData.testData
+        .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+        .limit(3)
+        .write
+        .mode("append")
+        .parquet(testPath)
+      hyperspace.refreshIndex(indexConfig.indexName, REFRESH_MODE_INCREMENTAL)
+      val newIndexCount = spark.read
+        .parquet(s"$systemPath/index/${IndexConstants.INDEX_VERSION_DIRECTORY_PREFIX}=1")
+        .count()
+
+      // Check if index got updated.
+      assert(newIndexCount == 3)
+
+      hyperspace.optimizeIndex(indexConfig.indexName)
+
+      // Check if latest log file is updated with newly created index files.
+      validateMetadata("index", Set("v__=2"))
```

Comment:
Do we have a test where we have index data in multiple locations (v__=0, v__=1)?

Reply:
thanks @imback82 , added test